gensim/test/test_similarities.py

```diff
@@ -155,12 +169,7 @@ def testPersistency(self):
             return
 
         fname = get_tmpfile('gensim_similarities.tst.pkl')
-        if self.cls == similarities.Similarity:
-            index = self.cls(None, corpus, num_features=len(dictionary), shardsize=5)
-        elif self.cls == similarities.WmdSimilarity:
-            index = self.cls(texts, self.w2v_model)
-        else:
-            index = self.cls(corpus, num_features=len(dictionary))
+        index = self.factoryMethod()
```

Comment:
are you sure that this an equivalent?

Reply:
Yes, this exact if-then-else statement currently appears at six places in `test_similarities.py` at `gensim/develop`. Rather than adding two new lines for `SoftCosineSimilarity` at each of the five locations, I decided to refactor it into a factory method.

Reply:
Should I make this a separate pull request?

Reply:
No (because this is really small change).

Reply:
Ok, I was more concerned with the fact that this pull request may take a little longer to finish up, whereas this single refactoring is independent and could be incorporated fast.